### PR TITLE
Taniyah l jackson/pinned comments on articles

### PR DESCRIPTION
**Pull request by Taniyah-L-Jackson**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, as well
as any relevant images for UI changes._

## Added tests?

- [ ] yes
- [ ] no, because they aren't needed
- [x] no, because I need help

## Added to documentation?

- [ ] docs.forem.com
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by Taniyah-L-Jackson** [comment]
So far, I noticed that that article model has a profile_pins relation and I'm wondering if that counts toward the profile_pin model. Assuming so, this is my outline so far.

File: app/models/profile_pin.rb [hunk 1]
```diff
@@ -1,21 +1,40 @@
-class ProfilePin < ApplicationRecord
+class ProfilePin < ApplicationRecord
```
Comment (citizen428) [inline app/models/profile_pin.rb:1]: **nitpick:** Some extra whitespace snuck in here.

File: app/models/profile_pin.rb [hunk 2]
```diff
@@ -1,21 +1,40 @@
...
+  validate :pin_comment_to_article
   validate :pinnable_belongs_to_profile
+  validate :article_to_article
 
   private
 
+  def get_profile_pins
+
+    #get all profile pins where an article has pinned comments
+    article.profile_pins.where("profile_id == ? AND profile_type == 'Article' AND pinnable_type == 'Comment'", params[:article.profile_id])
```
Comment (citizen428) [inline app/models/profile_pin.rb:20]: **suggestion:** We tend to use the ActiveRecord DSL as much as possible without explicity writing SQL snippets:

```suggestion
    article.profile_pins.where(profile_id: params[:article.profile_id], profile_type: Article, pinnable_type: Comment)
```

File: app/models/profile_pin.rb [hunk 3]
```diff
@@ -1,21 +1,40 @@
...
+  validates :pinnable_type, inclusion: { in: %w[Article Comment] } # Future could be comments, etc.
   validate :only_five_pins_per_profile, on: :create
+  validate :pin_comment_to_article
   validate :pinnable_belongs_to_profile
+  validate :article_to_article
 
   private
 
+  def get_profile_pins
+
```
Comment (citizen428) [inline app/models/profile_pin.rb:18]: **nitpick:** An extra empty line snuck in here.

File: app/models/profile_pin.rb [hunk 4]
```diff
@@ -1,21 +1,40 @@
...
+    #get all profile pins where an article has pinned comments
```
Comment (citizen428) [inline app/models/profile_pin.rb:19]: **suggestion (non-blocking):** I don't think this and the next comment add too much, we could remove them.

File: app/models/profile_pin.rb [hunk 5]
```diff
@@ -1,21 +1,40 @@
...
   def only_five_pins_per_profile
     errors.add(:base, "cannot have more than five total pinned posts") if profile.profile_pins.size > 4
   end
 
   def pinnable_belongs_to_profile
     errors.add(:pinnable_id, "must have proper permissions for pin") if pinnable.user_id != profile_id
   end
+
+  # If profile_type && pinnable_type = Article
+  def article_to_article
```
Comment (citizen428) [inline app/models/profile_pin.rb:36]: **suggestion (non-blocking):** To be more consistent with the rest of our codebase, this should use a guard clause:

```ruby
def article_to_article
  return unless pinnable_type == profile_type
  
  errors.add(:parent_type, "cannot pin articles to each other") 
end
```

**Review by citizen428 — COMMENTED** [review]
Thanks a lot for your contribution @Taniyah-L-Jackson. I just left a few minor comments indicating changes so this PR is a bit more in line with how we do things elsewhere.

File: app/models/profile_pin.rb (hunk 1 above)
Comment (benhalpern) [inline app/models/profile_pin.rb:1]:
```suggestion
class ProfilePin < ApplicationRecord
```

**Comment by Taniyah-L-Jackson** [comment]
Evening! Thank you @citizen428 and @benhalpern for your help. The file has been updated now and I'm hoping that the changes are logged.

**Review by citizen428 — APPROVED** [review]

**Comment by msarit** [comment]
Hey @Taniyah-L-Jackson! 👋🏾 
Just drawing your attention to this PR; there are some conflicts to resolve.
Once they are done, we'll work to get a second approval and have it merged. 🎉

**Comment by citizen428** [comment]
This also seems like a duplicate of #10503, so I'll close that one for now.

**Comment by msarit** [comment]
Hey @Taniyah-L-Jackson! 👋🏾
There are some failing CI checks (code-climate and Travis); please resolve! 🙏🏾 
Let us know if you need help! 😄

**Comment by msarit** [comment]
Pinging @Taniyah-L-Jackson once more, but I'm thinking it's safe to say this PR is not actively being worked on?
@jgaskins @rhymes

**Comment by Taniyah-L-Jackson** [comment]
> Pinging @Taniyah-L-Jackson once more, but I'm thinking it's safe to say this PR is not actively being worked on?
> @jgaskins @rhymes 

Yes

**Comment by msarit** [comment]
Thank you @Taniyah-L-Jackson 😄  Hope you keep contributing in the future. ❤️

**Pull request closed without merging**